File: src/test/java/javabot/operations/TellOperationTest.java
package javabot.operations;

import com.antwerkz.sofia.Sofia;
import javabot.BaseMessagingTest;
import javabot.dao.FactoidDao;
import javabot.dao.LogsDao;
import javabot.model.Logs;
import javabot.model.criteria.LogsCriteria;
import org.testng.Assert;
import org.testng.annotations.Test;

import javax.inject.Inject;

import static java.lang.String.*;

@Test
public class TellOperationTest extends BaseMessagingTest {
    @Inject
    private FactoidDao dao;

    @Inject
    private LogsDao logsDao;

    public void shortcut() {
        logsDao.deleteAllForChannel(getJavabotChannel().getName());
        final String nick = getJavabot().getNick();
        dao.delete(nick, "shortcut");
        try {
            final String message = "I'm a shortcut response";
            testMessage("~shortcut is <reply>" + message, ok);
            testMessage(format("~~ %s shortcut", getTestUser()),
                        format("%s, %s", getTestUser(), message));
            LogsCriteria criteria = new LogsCriteria(getDatastore());
            criteria.message(format("%s, %s", getTestUser().getNick(), message));
            final Logs logs = criteria.query().get();
            Assert.assertEquals(logs.getNick(), getJavabot().getNick());
        } finally {
            dao.delete(nick, "shortcut");
        }
    }

    public void unknownTell() {
        dao.delete(getJavabot().getNick(), "shortcut");
        testMessage(format("~~ %s shortcut", getTestUser()), Sofia.unhandledMessage(getTestUser().getNick()));
    }
}

File: src/test/kotlin/javabot/operations/TellOperationTest.kt
package javabot.operations

import com.antwerkz.sofia.Sofia
import javabot.BaseTest
import javabot.dao.FactoidDao
import javabot.model.criteria.LogsCriteria
import org.testng.Assert
import org.testng.annotations.Test
import javax.inject.Inject

@Test
public class TellOperationTest : BaseTest() {
    @Inject
    protected lateinit var dao: FactoidDao
    @Inject
    private lateinit var operation: GetFactoidOperation

    public fun shortcut() {
        logsDao.deleteAllForChannel(testChannel.name)
        val nick = bot.get().getNick()
        dao.delete(nick, "shortcut")
        try {
            val message = "I'm a shortcut response"
            var response = operation.handleMessage(message("shortcut is <reply>" + message))
            Assert.assertEquals(response[0].value, ok)
            response = operation.handleMessage(message("~ ${testUser} shortcut"))
            Assert.assertEquals(response[0].value, "${testUser}, ${message}")
            val criteria = LogsCriteria(datastore)
            criteria.message("${testUser.nick}, ${message}")
            val logs = criteria.query().get()
            Assert.assertEquals(logs.nick, bot.get().getNick())
        } finally {
            dao.delete(nick, "shortcut")
        }
    }

    public fun unknownTell() {
        dao.delete(bot.get().getNick(), "shortcut")
        var response = operation.handleMessage(message("~ ${testUser} shortcut"))
        Assert.assertEquals(response[0].value, Sofia.unhandledMessage(testUser.nick))
    }
}
